tools/cellxgene_census_builder/src/cellxgene_census_builder/build_state.py

```diff
@@ -20,38 +20,70 @@
     #
     # Paths and census version name determined by spec.
-    "cellxgene_census_S3_path": "s3://cellxgene-data-public/cell-census",
-    "logs_S3_path": "s3://cellxgene-data-public-logs/builder",
-    "build_tag": datetime.now().astimezone().date().isoformat(),
+    cellxgene_census_S3_path: str = field(default="s3://cellxgene-data-public/cell-census")
+    cellxgene_census_S3_remote_path: str = field(default=None)
+    logs_S3_path: str = field(default="s3://cellxgene-data-public-logs/builder")
+    build_tag: str = field(default=datetime.now().astimezone().date().isoformat())
     #
     # Default multi-process. Memory scaling based on empirical tests.
-    "multi_process": True,
-    "max_workers": 2 + int(psutil.virtual_memory().total / (96 * 1024**3)),
+    multi_process: bool = field(converter=bool, default=True)
+    max_workers: int = field(default=2 + int(psutil.virtual_memory().total / (96 * 1024**3)))
     #
     # Host minimum resource validation
-    "host_validation_disable": False,  # if True, host validation checks will be skipped
-    "host_validation_min_physical_memory": 512 * 1024**3,  # 512GiB
-    "host_validation_min_swap_memory": 2 * 1024**4,  # 2TiB
-    "host_validation_min_free_disk_space": 1 * 1024**4,  # 1 TiB
+    host_validation_disable: int = field(
+        converter=bool, default=False
+    )  # if True, host validation checks will be skipped
+    host_validation_min_physical_memory: int = field(converter=int, default=512 * 1024**3)  # 512Gi
+    host_validation_min_swap_memory: int = field(converter=int, default=2 * 1024**4)  # 2Ti
+    host_validation_min_free_disk_space: int = field(converter=int, default=1 * 1024**4)  # 1 Ti
     #
     # Release clean up
-    "release_cleanup_days": 32,  # Census builds older than this are deleted
+    release_cleanup_days: int = field(converter=int, default=32)  # Census builds older than this are delete)
     #
     # For testing convenience only
-    "manifest": None,
-    "test_first_n": None,
-}
+    manifest: str = field(default=None)
+    test_first_n: int = field(converter=int, default=None)
+
+    @classmethod
+    def load(cls, file: Union[str, os.PathLike[str], io.TextIOBase]) -> Self:
+        if isinstance(file, (str, os.PathLike)):
+            with open(file) as f:
+                user_config = yaml.safe_load(f)
+        else:
+            user_config = yaml.safe_load(file)
+
+        # Empty YAML config file is legal
+        if user_config is None:
+            user_config = {}
+
+        # But we only understand a top-level dictionary (e.g., no lists, etc.)
+        if not isinstance(user_config, dict):
+            raise TypeError("YAML config file malformed - expected top-level dictionary")
+
+        return cls(**user_config)
+
+    @classmethod
+    def load_from_env_vars(cls) -> Self:
+        config = cls()
+        for fld in fields(CensusBuildConfig):
+            if fld.name.upper() in os.environ:
```

Comment:
Question: do we want to prefix the ENV vars with something so there is less chance for conflict with other packages?   Names like VERBOSE are pretty common.

E.g., CENSUS_BUILD_VERBOSE?

Reply:
Probably a good idea.

Reply:
Any update on this?

Reply:
I don't think you can assume that the docker image will only ever have just the Builder in it.  There are both dependent packages which use environment variables, and other code we may add to be part of the build process.  I think for best practices, I would clean up this namespace.

Reply:
Added a `CENSUS_BUILD` prefix. This will require some changes in the SFN, but they should be straightforward.